share/nixos-shell.nix

```diff
@@ -0,0 +1,26 @@
+{
+  nixpkgs ? <nixpkgs>
+, system ? builtins.currentSystem
+, configuration ? <nixos-config>
+
+, flakeUri ? null
+, flakeAttr ? null
+}:
+let 
+  nixos-shell-module = import ./modules/nixos-shell.nix;
+
+  flake = builtins.getFlake flakeUri;
+  flakeSystem = flake.outputs.packages."${system}".nixosConfigurations."${flakeAttr}" or flake.outputs.nixosConfigurations."${flakeAttr}";
```

Comment:
Should we also lookup nixos modules? nixos-shell could be quite handy for testing them.

Reply:
This would be very useful for me. I want to test a nixosModule from someone else's flake.

Reply:
I think it should be treated the same way we currently treat modules being passed as the first argument to `nixos-shell` when not using flakes.

Reply:
But this can be also implemented in a different PR.